### PR TITLE
feature (filters): Add some filters for better control …

### PR DESCRIPTION
**Pull request by Camwyn**
of the event/vendor/organizer link targets.

Standardize the way we create those links (HTML) for easier maintenance.

Note https://github.com/mt-support/tribe-ext-external-links-in-new-tab/pull/1 relies on these filter changes, but this is safe to release without EXT-151 going live.

[TEC-3402]

[TEC-3402]: https://moderntribe.atlassian.net/browse/TEC-3402

### CODE REVIEW

File: src/functions/template-tags/link.php [hunk 1]
```diff
@@ -357,14 +357,19 @@ function tribe_get_event_link( $post_id = null, $full_link = false ) {
 	 * @return string $html
 	 */
 	function tribe_get_event_website_link( $event = null, $label = null ) {
-		$url = tribe_get_event_website_url( $event );
+		$url    = tribe_get_event_website_url( $event );
+		$target = apply_filters( 'tribe_get_event_website_link_target', '_self' );
+		$rel    = ( '_blank' === $target  ) ? 'noopener noreferrer' : 'external';
```
Comment (tr1b0t) [inline src/functions/template-tags/link.php:362]: :no_entry_sign: Expected 1 space before close parenthesis; 2 found

<sub>hash: 47f6ceaf4d861d356be1959ecea5c0ab</sub>

File: src/functions/template-tags/organizer.php [hunk 2]
```diff
@@ -265,11 +265,25 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
 
 		$org_id = tribe_get_organizer_id( $postId );
 		if ( class_exists( 'Tribe__Events__Pro__Main' ) && get_post_status( $org_id ) == 'publish' ) {
-			$url = esc_url_raw( get_permalink( $org_id ) );
+			$url    = esc_url_raw( get_permalink( $org_id ) );
+			$target = apply_filters( 'tribe_get_event_organizer_link_target', '_self' );
+			$rel    = ( '_blank' === $target  ) ? 'noopener noreferrer' : '';
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:270]: :no_entry_sign: Expected 1 space before close parenthesis; 2 found

<sub>hash: 88797b0ec539d840a6a44915e3b5c8d2</sub>

File: src/functions/template-tags/organizer.php [hunk 3]
```diff
@@ -265,11 +265,25 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
...
+
 			if ( $full_link ) {
 				$name = tribe_get_organizer( $org_id );
-				$attr_title = the_title_attribute( array( 'post' => $org_id, 'echo' => false ) );
-				$link = ! empty( $url ) && ! empty( $name ) ? '<a href="' . esc_url( $url ) . '" title="'.$attr_title.'">' . $name . '</a>' : false;
+
+				if ( empty( $url ) || empty( $name ) ) {
+					$link = false;
+				} else {
+					$link  = sprintf(
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:278]: :warning: Equals sign not aligned correctly; expected 1 space but found 2 spaces

<sub>hash: a6924ce0263897116e42d47446f86988</sub>

File: src/functions/template-tags/organizer.php [hunk 4]
```diff
@@ -265,11 +265,25 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
...
+						'<a href="%s" title="%s" target="%s" rel="%s">%s</a>',
+						esc_url( $url ),
+						the_title_attribute( array( 'post' => $org_id, 'echo' => false ) ),
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:281]: :no_entry_sign: When a multi-item array uses associative keys, each value should start on a new line.

<sub>hash: cd143bcbffff45ed45a28d972515e358</sub>

File: src/functions/template-tags/organizer.php [hunk 5]
```diff
@@ -332,19 +347,25 @@ function tribe_get_organizer_website_url( $postId = null ) {
 	function tribe_get_organizer_website_link( $post_id = null, $label = null ) {
 		$post_id = tribe_get_organizer_id( $post_id );
 		$url     = tribe_get_event_meta( $post_id, '_OrganizerWebsite', true );
+		$target = apply_filters( 'tribe_get_event_organizer_link_target', '_self' );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:350]: :warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space

<sub>hash: cdbd897f1fffc2f522ce229c1a872ad3</sub>

File: src/functions/template-tags/organizer.php [hunk 6]
```diff
@@ -332,19 +347,25 @@ function tribe_get_organizer_website_url( $postId = null ) {
...
+		$rel    = ( '_blank' === $target  ) ? 'noopener noreferrer' : 'external';
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:351]: * :warning: Equals sign not aligned with surrounding assignments; expected 5 spaces but found 4 spaces
* :no_entry_sign: Expected 1 space before close parenthesis; 2 found

<sub>hash: c05f003d0e3f70fb8dd54dcaff816f99</sub>

File: src/functions/template-tags/organizer.php [hunk 7]
```diff
@@ -332,19 +347,25 @@ function tribe_get_organizer_website_url( $postId = null ) {
...
+		$label = apply_filters( 'tribe_get_organizer_website_link_label', $label );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:352]: :warning: Equals sign not aligned with surrounding assignments; expected 3 spaces but found 1 space

<sub>hash: c446801cfad2c1a122f24690361d2c48</sub>

File: src/functions/template-tags/venue.php [hunk 8]
```diff
@@ -736,6 +736,7 @@ function tribe_get_venue_website_link( $post_id = null, $label = null ) {
 			 * @param int $post_id The venue ID.
 			 */
 			$website_link_target = apply_filters( 'tribe_get_venue_website_link_target', '_self', $post_id );
+			$rel                 = ( '_blank' === $website_link_target  ) ? 'noopener noreferrer' : 'external';
```
Comment (tr1b0t) [inline src/functions/template-tags/venue.php:739]: :no_entry_sign: Expected 1 space before close parenthesis; 2 found

<sub>hash: 595af3c721a81375a221df9748fadcf7</sub>

File: src/functions/template-tags/organizer.php [hunk 9]
```diff
@@ -265,11 +265,28 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
...
-				$attr_title = the_title_attribute( array( 'post' => $org_id, 'echo' => false ) );
-				$link = ! empty( $url ) && ! empty( $name ) ? '<a href="' . esc_url( $url ) . '" title="'.$attr_title.'">' . $name . '</a>' : false;
+
+				if ( empty( $url ) || empty( $name ) ) {
+					$link = false;
+				} else {
+					$link = sprintf(
+						'<a href="%s" title="%s" target="%s" rel="%s">%s</a>',
+						esc_url( $url ),
+						the_title_attribute( [
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:281]: :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 43db05976a121127bfa4200aada63678</sub>

File: src/functions/template-tags/organizer.php [hunk 10]
```diff
@@ -265,11 +265,28 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
...
+							'post' => $org_id,
+							'echo' => false
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:283]: * :no_entry_sign: Each array item in a multi-line array declaration must end in a comma
* :no_entry_sign: Each line in an array declaration must end in a comma

<sub>hash: 6fa8266b33a638b06d9c38236bd6920c</sub>

File: src/functions/template-tags/organizer.php [hunk 11]
```diff
@@ -265,11 +265,28 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
...
+						] ),
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:284]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 792c23f6abb49617d728f5cfb9001554</sub>

File: src/functions/template-tags/link.php [hunk 12]
```diff
@@ -357,14 +357,19 @@ function tribe_get_event_link( $post_id = null, $full_link = false ) {
 	 * @return string $html
 	 */
 	function tribe_get_event_website_link( $event = null, $label = null ) {
-		$url = tribe_get_event_website_url( $event );
+		$url    = tribe_get_event_website_url( $event );
+		$target = apply_filters( 'tribe_get_event_website_link_target', '_self' );
```
Comment (lucatume) [inline src/functions/template-tags/link.php:361]: Add filter doc block.

File: src/functions/template-tags/organizer.php [hunk 13]
```diff
@@ -265,11 +265,30 @@ function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = fa
 
 		$org_id = tribe_get_organizer_id( $postId );
 		if ( class_exists( 'Tribe__Events__Pro__Main' ) && get_post_status( $org_id ) == 'publish' ) {
-			$url = esc_url_raw( get_permalink( $org_id ) );
+			$url    = esc_url_raw( get_permalink( $org_id ) );
+			$target = apply_filters( 'tribe_get_event_organizer_link_target', '_self' );
```
Comment (lucatume) [inline src/functions/template-tags/organizer.php:269]: Add filter doc block.

File: src/functions/template-tags/organizer.php [hunk 14]
```diff
@@ -332,19 +352,25 @@ function tribe_get_organizer_website_url( $postId = null ) {
 	function tribe_get_organizer_website_link( $post_id = null, $label = null ) {
 		$post_id = tribe_get_organizer_id( $post_id );
 		$url     = tribe_get_event_meta( $post_id, '_OrganizerWebsite', true );
+		$target  = apply_filters( 'tribe_get_event_organizer_link_target', '_self' );
```
Comment (lucatume) [inline src/functions/template-tags/organizer.php:355]: Add filter doc block.

File: src/functions/template-tags/organizer.php [hunk 15]
```diff
@@ -332,19 +352,25 @@ function tribe_get_organizer_website_url( $postId = null ) {
...
+		$rel     = ( '_blank' === $target ) ? 'noopener noreferrer' : 'external';
+		$label   = apply_filters( 'tribe_get_organizer_website_link_label', $label );
```
Comment (lucatume) [inline src/functions/template-tags/organizer.php:357]: Add filter doc block.

File: src/functions/template-tags/link.php [hunk 16]
```diff
@@ -357,14 +357,19 @@ function tribe_get_event_link( $post_id = null, $full_link = false ) {
...
+		$rel    = ( '_blank' === $target ) ? 'noopener noreferrer' : 'external';
+
 		if ( ! empty( $url ) ) {
 			$label = is_null( $label ) ? $url : $label;
+			$label = apply_filters( 'tribe_get_event_website_link_label', $label );
```
Comment (lucatume) [inline src/functions/template-tags/link.php:366]: Add filter doc block.

File: src/functions/template-tags/link.php [hunk 17]
```diff
@@ -357,19 +357,45 @@ function tribe_get_event_link( $post_id = null, $full_link = false ) {
 	 * @return string $html
 	 */
 	function tribe_get_event_website_link( $event = null, $label = null ) {
-		$url = tribe_get_event_website_url( $event );
+		$url    = tribe_get_event_website_url( $event );
```
Comment (tr1b0t) [inline src/functions/template-tags/link.php:360]: :warning: Equals sign not aligned correctly; expected 1 space but found 4 spaces

<sub>hash: 25a3f2a14d8068a53de5774f262dd70a</sub>

File: src/functions/template-tags/organizer.php [hunk 18]
```diff
@@ -253,34 +253,64 @@ function tribe_get_organizer_email( $postId = null, $antispambot = true ) {
 	 *
 	 * @param int  $postId    Can supply either event id or organizer id, if none specified, current post is used
 	 * @param bool $full_link If true outputs a complete HTML <a> link, otherwise only the URL is output
-	 * @param bool $echo      If true, echo the link, otherwise return
+	 * @param bool $echo      Deprecated. If true, echo the link, otherwise return
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:256]: :no_entry_sign: Parameter comment must end with a full stop

<sub>hash: 6ed5bf131da02c22421fc4a1438a9372</sub>

File: src/functions/template-tags/organizer.php [hunk 19]
```diff
@@ -253,34 +253,64 @@ function tribe_get_organizer_email( $postId = null, $antispambot = true ) {
...
 	 *
 	 * @return string Organizer Name and Url
 	 */
 	function tribe_get_organizer_link( $postId = null, $full_link = true, $echo = false ) {
 
 		// As of TEC 4.0 this argument is deprecated
 		// If needed precede the call to this function with echo
-		if ( $echo != false ) _deprecated_argument( __FUNCTION__, '4.0' );
+		if ( $echo != false ) _deprecated_argument( __FUNCTION__, '4.0', 'As of TEC 4.0 this argument is deprecated. If needed, precede the call to this function with echo' );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:264]: * :no_entry_sign: Inline control structures are not allowed
* :no_entry_sign: Use Yoda Condition checks, you must.

<sub>hash: 46cb13082d5a3882888b9037d862af1f</sub>

File: src/functions/template-tags/organizer.php [hunk 20]
```diff
@@ -253,34 +253,64 @@ function tribe_get_organizer_email( $postId = null, $antispambot = true ) {
...
-			} else {
-				return apply_filters( 'tribe_get_organizer_link', $link, $postId, $full_link, $url );
+			/**
+			 * Filter the organizer link HTML
+			 *
+			 * @since 4.0
+			 *
+			 * @param string the link HTML.
+			 */
+			return apply_filters( 'tribe_get_organizer_link', $link, $postId, $full_link, $url );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:310]: :no_entry_sign: Variable "$postId" is not in valid snake_case format, try "$post_id"

<sub>hash: e15475b69f49b995bbf10e075a79d8f6</sub>

File: src/functions/template-tags/organizer.php [hunk 21]
```diff
@@ -332,24 +362,53 @@ function tribe_get_organizer_website_url( $postId = null ) {
...
+		$rel    = ( '_blank' === $target ) ? 'noopener noreferrer' : 'external';
+
+		/**
+		 * Filter the organizer link label
+		 *
+		 * @since TBD
+		 *
+		 * @param string the link label/text.
+		 */
+		$label  = apply_filters( 'tribe_get_organizer_website_link_label', $label );
```
Comment (tr1b0t) [inline src/functions/template-tags/organizer.php:383]: :warning: Equals sign not aligned correctly; expected 1 space but found 2 spaces

<sub>hash: a2d01cb6de424d1d146a87c9ab986a3d</sub>

**Pull request merged**